### PR TITLE
starship: add page

### PR DESCRIPTION
**Pull request by navarroaxel**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/starship.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs/>.
+
+- Print the shell function used to execute starship:
+
+`starship init {{bash|fish|zsh}`
```
Comment (bl-ue) [inline pages/common/starship.md:8]: Is there a reason why you didn't include all supported shells?
```suggestion
`starship init {{bash|zsh|fish|powershell|ion|elvish|tcsh}}`
```
(and is there a reason why you ended the token syntax with only one `}`? 😄)

File: pages/common/starship.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+`starship configure`
+
+- Create a pre-populated GitHub issue with information about your configuration:
+
+`starship bug-report`
+
+- Print a completion script:
+
+`starship completions {{bash|fish|zsh}}`
```
Comment (bl-ue) [inline pages/common/starship.md:32]: Same here. Why only the first three?
```suggestion
`starship completions {{bash|fish|zsh}}`
```

File: pages/common/starship.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs/>.
```
Comment (marchersimon) [inline pages/common/starship.md:4]:
```suggestion
> More information: <https://starship.rs>.
```

File: pages/common/starship.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+- Print the computed starship configuration (use `--default` to print default configuration instead):
+
+`starship print-config`
+
+- List supported modules:
+
+`starship module --list`
+
+- Edit the starship configuration:
```
Comment (marchersimon) [inline pages/common/starship.md:22]: Does it use the systems default or the one in `$EDITOR`?

Comment (bl-ue) [reply pages/common/starship.md:22]: It uses `$EDITOR` if it's set, otherwise the system's default (`vim` for me, on macOS).

Comment (marchersimon) [reply pages/common/starship.md:22]: Should we maybe update the description to `Edit the starship configuration in the default editor`?

Comment (bl-ue) [reply pages/common/starship.md:22]: Sounds good to me. 👍🏻

Comment (navarroaxel) [reply pages/common/starship.md:22]:
```suggestion
- Edit the starship configuration in the default editor:
```

File: pages/common/starship.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs>.
+
+- Print the shell function used to execute starship:
+
+`starship init {{bash|elvish|fish|ion|powershell|tcsh|zsh}}`
```
Comment (marchersimon) [inline pages/common/starship.md:8]: Isn't the shell function to execute starship `starship`?

Comment (navarroaxel) [reply pages/common/starship.md:8]: Yep, but the installation is different for every shell.
```
eval "$(starship init bash)"
Invoke-Expression (&starship init powershell)
eval (starship init elvish)
eval `starship init tcsh`
```

Maybe I can create a subcommand page for `starship-init` similar to `gh-completion`

Comment (bl-ue) [reply pages/common/starship.md:8]: Not a bad idea, but that's for another PR.

File: pages/common/starship.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs>.
+
+- Print the shell function used to execute starship:
```
Comment (bl-ue) [inline pages/common/starship.md:6]: I like the following:
```suggestion
- Generate starship integration code for the current shell and print it to stdout:
```

Comment (navarroaxel) [reply pages/common/starship.md:6]: Actually it doesn't generate nothing, only prints the selected script.

Comment (bl-ue) [reply pages/common/starship.md:6]: But I don't think that's a necessary detail—usually "generate" refers to this type of thing (if if currently is it just a dump of a static file's contents).

Comment (marchersimon) [reply pages/common/starship.md:6]: To me `generate` also sounds like it is creating the code when you execute the command. How about:
```suggestion
- Print starship integration code for the specified shell:
```

Comment (sbrl) [reply pages/common/starship.md:6]: I think I like @marchersimon's best, but with a slightly grammatical tweak:
```suggestion
- Print the starship integration code for the specified shell:
```

File: pages/common/starship.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+- List supported modules:
+
+`starship module --list`
+
+- Edit the starship configuration in the default editor:
+
+`starship configure`
+
+- Create a pre-populated GitHub issue with information about your configuration:
```
Comment (bl-ue) [inline pages/common/starship.md:26]:
```suggestion
- Create a bug report GitHub issue pre-populated with information about your system and starship configuration:
```

File: pages/common/starship.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs>.
+
+- Print the starship integration code for the specified shell:
+
+`starship init {{bash|elvish|fish|ion|powershell|tcsh|zsh}}`
+
+- Explain the current showing modules:
```
Comment (marchersimon) [inline pages/common/starship.md:10]:
```suggestion
- Explain the currently showing modules:
```
Except if they are showing the current of course 😅

File: pages/common/starship.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+- Edit the starship configuration in the default editor:
+
+`starship configure`
+
+- Create a pre-populated GitHub issue with information about your configuration:
+
+`starship bug-report`
+
+- Print a completion script:
```
Comment (marchersimon) [inline pages/common/starship.md:30]: To keep it consistent with the first example:
```suggestion
- Print the completion script for the specified shell:
```

**Review by CleanMachine1 — CHANGES_REQUESTED** [review]
I didn't send my review..

File: pages/common/starship.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+- Print the computed starship configuration (use `--default` to print default configuration instead):
+
+`starship print-config`
+
+- List supported modules:
+
+`starship module --list`
+
+- Edit the starship configuration in the default editor:
```
[the remark this answers is not in the document]
Comment (navarroaxel) [reply pages/common/starship.md:22]: in all pages, like `git-config` we use `in the default editor`.

**Review by marchersimon — APPROVED** [review]
Looks good, thanks

File: pages/common/starship.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
+# starship
+
+> The minimal, blazing-fast, and infinitely customizable prompt for any shell.
+> More information: <https://starship.rs>.
+
+- Print the starship integration code for the specified shell:
+
+`starship init {{bash|elvish|fish|ion|powershell|tcsh|zsh}}`
+
+- Explain the currently showing modules:
```
Comment (bl-ue) [inline pages/common/starship.md:10]: How about this, instead?
```suggestion
- Break down the current prompt and explain each part of it, as well as the time taken to render it:
```

**Review by sbrl — APPROVED** [review]
Thanks, @navarroaxel!

**Pull request merged**